opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -250,6 +250,9 @@ namespace detail {
         const bool converged = asImpl().getConvergence(dt, iteration);
         const bool must_solve = (iteration < nonlinear_solver.minIter()) || (!converged);
         if (must_solve) {
+            // enable single precision for solvers when dt is smaller then 20 days
+            residual_.singlePrecision = (unit::convert::to(dt, unit::day) < 20.) ;
```

Comment:
Just out of curiosity, what is the theory here? Why with small `dt` single precision is desired?